executor/instruction/iterator.rs

```diff
@@ -125,10 +115,25 @@ impl<I: for<'a> LendingIterator<Item<'a> = TupleResult<'static>> + TupleSeekable
         // TODO: this is close to a copy-paste of the Seek() implementation for seekable KMergeBy<I>
         self.iterators = mem::take(&mut self.iterators)
             .drain()
-            .map(|mut it| it.iter.seek(target).map(|_| it))
+            .filter_map(|mut it| {
+                if let Some(Ok(item)) = it.iter.peek() {
+                    if item < target {
+                        if let Err(err) = it.iter.seek(target) {
+                            return Some(Err(err));
+                        }
+                    }
+                }
+                it.iter.peek().is_some().then_some(Ok::<_, Box<ConceptReadError>>(it))
+            })
```

Comment:
Check if the merge targets are actually behind the seek target. It's valid for some to be ahead.